### PR TITLE
Chromeless mode

### PR DESCRIPTION
**Pull request by helbertgs**
## Goal

Introduce `Chromeless` mode to remove user interaction and visual elements.

## How to test

Add the code on ViewController.swift#26:

```swift
DispatchQueue.main.asyncAfter(deadline: .now()+5) {
    self.player.chromelessMode = true
}

DispatchQueue.main.asyncAfter(deadline: .now()+10) {
    self.player.chromelessMode = false
}
```

- At 00:05s the interactions with playback will be disabled and the visual elements will be hide.
- At 00:10s the interactions with playback will be enabled.

### CODE REVIEW

**Review by geisonfgf — APPROVED** [review]

File: Tests/Clappr_tvOS_Tests/PlayerTests.swift [hunk 1]
```diff
@@ -282,6 +282,72 @@ class PlayerTests: QuickSpec {
                     }
                 }
             }
+
+            describe("#chromeless") {
+                context("when enter chromeless mode is call") {
```
Comment (BruunoMM) [inline Tests/Clappr_tvOS_Tests/PlayerTests.swift:287]:
```suggestion
                context("when enterChromelessMode is called") {
```

File: Tests/Clappr_tvOS_Tests/PlayerTests.swift [hunk 2]
```diff
@@ -282,6 +282,72 @@ class PlayerTests: QuickSpec {
...
+                    it("disables the user interaction") {
+                        let player = Player()
+
+                        player.enterChromelessMode()
+
+                        expect(player.view.isUserInteractionEnabled).to(beFalse())
+                    }
+                }
+
+                context("when exit chromeless mode is call") {
```
Comment (BruunoMM) [inline Tests/Clappr_tvOS_Tests/PlayerTests.swift:318]:
```suggestion
                context("when exitChromelessMode is called") {
```

File: Tests/Clappr_tvOS_Tests/PlayerTests.swift (hunk 1 above)
Comment (helbertgs) [reply Tests/Clappr_tvOS_Tests/PlayerTests.swift:287]: Done

File: Tests/Clappr_tvOS_Tests/PlayerTests.swift (hunk 2 above)
Comment (helbertgs) [reply Tests/Clappr_tvOS_Tests/PlayerTests.swift:318]: Done

File: Sources/Clappr_tvOS/Classes/Base/Player.swift [hunk 3]
```diff
@@ -313,6 +315,20 @@ open class Player: AVPlayerViewController {
...
+        view.isUserInteractionEnabled = false
+        activeContainer?.trigger(InternalEvent.didEnterChromelessMode.rawValue)
+    }
+
+    open func exitChromelessMode() {
+        isChromelessModeEnabled = false
+        showsPlaybackControls = true
+        view.isUserInteractionEnabled = true
+        activeContainer?.trigger(InternalEvent.didExitChromelessMode.rawValue)
+    }
```
Comment (fcapinho) [inline Sources/Clappr_tvOS/Classes/Base/Player.swift:330]: I recommend making this functions private and keeping all chromeless mode control on a property. 
This is closer to the chromeless mode control on Clappr and Clappr-Android: a dynamic option.

**Review by fcapinho — CHANGES_REQUESTED** [review]

File: Sources/Clappr_tvOS/Classes/Base/Player.swift (hunk 3 above)
Comment (helbertgs) [reply Sources/Clappr_tvOS/Classes/Base/Player.swift]: Done

File: Tests/Clappr_tvOS_Tests/PlayerTests.swift [hunk 4]
```diff
@@ -282,6 +282,72 @@ class PlayerTests: QuickSpec {
...
+                    }
+
+                    it("disables the user interaction") {
+                        let player = Player()
+
+                        player.enterChromelessMode()
+
+                        expect(player.view.isUserInteractionEnabled).to(beFalse())
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (helbertgs) [reply Tests/Clappr_tvOS_Tests/PlayerTests.swift]: Done

File: Tests/Clappr_tvOS_Tests/PlayerTests.swift [hunk 5]
```diff
@@ -282,6 +282,72 @@ class PlayerTests: QuickSpec {
                     }
                 }
             }
+
+            describe("#chromeless") {
```
[the remark this answers is not in the document]
Comment (helbertgs) [reply Tests/Clappr_tvOS_Tests/PlayerTests.swift]: Done

File: Sources/Clappr_tvOS/Classes/Base/Player.swift [hunk 6]
```diff
@@ -115,6 +115,16 @@ open class Player: AVPlayerViewController {
         }
     }
 
+    public var chromelessMode = false {
```
Comment (fcapinho) [inline Sources/Clappr_tvOS/Classes/Base/Player.swift:118]: Chromeless mode control is now split in two separate properties: `isChromeless` and `chromelessMode`. They reflect the same state but are not synchronized. This leads to inconsistencies in the code (e.g.: the check for `isChromeless` and not for `chromelessMode` in `willEnterForeground`).

I recommend keeping this state in a single place and making sure the `kChromeless` option received on the Player scope is only the initial value for this state.

**Review by fcapinho — CHANGES_REQUESTED** [review]

File: Sources/Clappr_tvOS/Classes/Base/Player.swift (hunk 6 above)
Comment (helbertgs) [reply Sources/Clappr_tvOS/Classes/Base/Player.swift]: We merge `isChromeless` with `chromelessMode` and now we are keeping this state in a single place.

**Review by fcapinho — APPROVED** [review]

**Review by BruunoMM — APPROVED** [review]

**Review by torives — APPROVED** [review]

**Pull request merged**